### PR TITLE
ci: add template workflows for a DRY centralize plugin CI

### PR DESCRIPTION
**Pull request by h1dden-da3m0n**
### Description

Once this is in and all org held plugins use them we have a central location to change things :tada: 

### Changes

* adds DRY workflow definitions that all org held plugin repos can use

### Issues

* n/a

### CODE REVIEW

**Comment by h1dden-da3m0n** [comment]
Note, I would like to add Renovate to this repo to keep the actions used in the DRY workflows up to date, but since it would pick up the sub modules too I would like to ask if we want to keep submodules within this repository?

File: .github/workflows/bump-version.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+          yq-version: v4.16.2
+
+      - name: Read current version
+        run: |-
+          echo "CURRENT_VERSION=$(yq eval '.version' build.yaml)"
+          echo "CURRENT_VERSION=$(yq eval '.version' build.yaml)" >> $GITHUB_ENV
+
+      - name: Set next version
+        run: |-
+          # ensure even x.y.z.b is supporter
```
Comment (crobibero) [inline .github/workflows/bump-version.yaml:36]:
```suggestion
          # ensure even x.y.z.b is superior
```

**Comment by crobibero** [comment]
> but since it would pick up the sub modules too I would like to ask if we want to keep submodules within this repository?

I was going to ask if we could auto update the submodules... so that might be ideal?

**Comment by h1dden-da3m0n** [comment]
yes it could, but what are those submodules even good for? do we use them anywhere?

**Comment by crobibero** [comment]
I use them locally when I need to update all the plugins :)

File: .github/workflows/bump-version.yaml (hunk 1 above)
Comment (h1dden-da3m0n) [reply .github/workflows/bump-version.yaml:36]: supported is what I wanted to write there :P

**Comment by h1dden-da3m0n** [comment]
okay that is actually an interesting way to 'catch them all'. are you able to direct commit to all of the repos from within the sub-modules?

**Comment by crobibero** [comment]
Sure can, just have to ensure that the origin is pointing to the plugin repo instead of meta-plugins

File: .github/renovate.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      },
+      {
+        "matchManagers": ["npm"],
+        "addLabels": ["npm"]
+      },
+      {
+        "matchManagers": ["git-submodules"],
+        "addLabels": ["git-submodules"],
+        "automerge": true,
+        "automergeType": "branch"
```
Comment (h1dden-da3m0n) [inline .github/renovate.json:19]: I assumed we want to auto merge submodule updates, so I configured them as such.

NOTE: https://docs.renovatebot.com/modules/manager/git-submodules/
> Git Submodules functionality is currently in beta testing so you must opt in to test it out.

**Review by crobibero — APPROVED** [review]
Approved pending @joshuaboniface review of `.github/workflows/publish.yaml `

**Comment by h1dden-da3m0n** [comment]
1 request: please squash merge this xD

**Comment by crobibero** [comment]
Is it possible for repos not in the org to use these workflows? If so, it would be nice to disable the asset upload to our repo server

**Comment by h1dden-da3m0n** [comment]
> Is it possible for repos not in the org to use these workflows? If so, it would be nice to disable the asset upload to our repo server

Yes you can use these workflows outside of the org as well of course!
I could add a `if` that checks if the repo calling the release workflow is called from a repo within the org and skip it in the case that it is not.

**Review by joshuaboniface — APPROVED** [review]

**Pull request merged**